### PR TITLE
Use openshift/router-haproxy image for haproxy

### PR DESCRIPTION
**Pull request by yboaron**
This change updates haproxy static pod manifest to use
openshift/router-haproxy instead of docker/haproxy

### CODE REVIEW

File: assets/files/etc/kubernetes/manifests/haproxy.yaml [hunk 1]
```diff
@@ -68,15 +68,33 @@ spec:
...
+            /usr/sbin/haproxy -W -db -f /etc/haproxy/haproxy.cfg  -p /var/lib/haproxy/run/haproxy.pid &
+        fi
+      }
+      /usr/sbin/haproxy -W -db -f /etc/haproxy/haproxy.cfg  -p /var/lib/haproxy/run/haproxy.pid &
+      while true; do
+          sleep 10
+          if  grep -q RELOAD /haproxy_reload/msgq; then
+              reload_haproxy
+              echo "" > /haproxy_reload/msgq
+          fi
```
Comment (russellb) [inline assets/files/etc/kubernetes/manifests/haproxy.yaml]: I think there’s a race condition here where a config change could get missed. A second config change could come in after a reload but before the msgq file is cleared. That change would be missed.

I know it seems incredibly unlikely. I can think of two options. 

1 - use locking on msgq as the shared resource. 

2 - treat msgq as a queue. Append a line for each reload request and only remove a line when reloading.

**Comment by russellb** [comment]
And if you make it a real queue maybe it shouldn’t be a file and should be a Unix domain socket, something that has an interface closer to what’s needed?

**Comment by yboaron** [comment]
@russellb , The race issue is a good point, I changed consumer (haproxy) poll interval to be smaller than half of producer (haproxy-monitor) poll interval, so we should be fine now with the race issue.
Performance wise it should be better to use something like unix domain socket instead of busy-wait polling.
Do u think it's acceptable if I'll send it in a follow-up patch?

**Comment by russellb** [comment]
I think the race is still theoretically possible, even if you adjust the timeouts.  I think it's worth investigating other approaches before merging.

File: assets/files/etc/kubernetes/manifests/haproxy.yaml [hunk 2]
```diff
@@ -108,8 +128,10 @@ spec:
       API_VIP="$(dig +noall +answer "api.${DOMAIN}" | awk '{print $NF}')"
 
       trap clean_prerouting_rules SIGINT SIGTERM
+      touch /haproxy_reload/msgq
+      chmod 777  /haproxy_reload/msgq
       while true; do
-          sleep 15
+          sleep 25
```
Comment (celebdor) [inline assets/files/etc/kubernetes/manifests/haproxy.yaml]: How's this related to the PR? What's the motivation for increasing the sleep?

**Comment by celebdor** [comment]
Agreed on locking or unix domain socket

**Comment by bcrochet** [comment]
This is obsolete and should be closed. @yboaron

**Comment by yboaron** [comment]
The haproxy static pod moved to MCO - we can delete this PR
10x @bcrochet

**Comment by yboaron** [comment]
The haproxy static pod moved to MCO - we can close this PR
10x @bcrochet

**Pull request closed without merging**